pyrobosim/pyrobosim/gui/main.py

```diff
@@ -318,3 +327,8 @@ def on_collision_polygon_toggle_click(self):
         """Callback to toggle collision polygons."""
         self.canvas.toggle_collision_polygons()
         self.canvas.draw_signal.emit()
+
+    def on_cancel_action_click(self):
+        """Callback to cancel any running action"""
+        [r.cancel_actions() for r in self.world.robots]
+        self.canvas.draw_signal.emit()
```

Comment:
Rather than canceling every robot's action, this should only cancel the active robot's action.

This class has a `get_current_robot()` method exactly for this; you can refer to other usages throughout this file.